### PR TITLE
connect to API pull request

### PR DESCRIPTION
**Pull request by theresetuyi**
## Bookstore: connect to API
Use the concept of state management.
Use store, actions and reducers in React.
Connect React and Redux.
Connect an existing API via Redux and use the data to display something in a UI.

### CODE REVIEW

File: src/BooksApi/BooksApi.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+const BASE_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/NmerLOYOolwhXAVjaTow/books';
+
+export const fetchBooks = async () => {
+  const response = await axios.get(`${BASE_URL}/books`);
+  return response.data;
+};
+
+export const addBook = async (book) => {
+  const response = await axios.post(`${BASE_URL}/books`, book);
+  return response.data;
```
Comment (Davidosky007) [inline src/BooksApi/BooksApi.js]: * I get this error in the console when I try to add a book, Kindly fix it as the API doesn't function like it should 
image

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
+  {
+    item_id: 'item3',
+    title: 'The Selfish Gene',
+    author: 'Richard Dawkins',
+    category: 'Nonfiction',
+  },
+];
 export const fetchBooksAsync = createAsyncThunk('books/fetchBooks', async () => {
   const response = await fetchBooks();
+  // console.log(response, 'this is api data');
```
Comment (Davidosky007) [inline src/redux/books/booksSlice.js]: * Kindly remove all commented code form your code base as it makes your code bulky

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @theresetuyi 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Store, actions and reducers in used 👍 
* Redux setup properly 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js [hunk 3]
```diff
@@ -1,18 +1,38 @@
...
+  },
+];
+export const getbooks = createAsyncThunk(
+  'books/getbooks',
+  async () => {
+    const response = await axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Oc9abMsHOhG37sSkzGFr/books/');
+    return response.data;
+  },
+);
 export const fetchBooksAsync = createAsyncThunk('books/fetchBooks', async () => {
```
Comment (FrederickMih) [inline src/redux/books/booksSlice.js]: - [ ] Kindly make sure that fetchBooksAsync actually updates the state of booksSlice. Right now I'm still unable to add a book. Only the default books from the books array are displayed. Kindly make sure it is possible to add a book.

[Screencast from 06-15-2023 02:56:32 PM.webm](https://github.com/theresetuyi/React-Bookstore/assets/63087491/24c9eeac-1688-4f02-a3d4-4e39ecaa730d)

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @theresetuyi , 


Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- No Linter failure  :+1: 
- PR is descriptive :+1: 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @theresetuyi , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done implementing the changes requested by the previous reviewer. ✔
- We can succesfully add a book now. ✔
- You successfully completed all the requirements.  🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**